src/main/rx/msp_override.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "platform.h"
+
+#if defined(USE_RX_MSP_OVERRIDE)
+
+#include "rx/msp_override.h"
+#include "rx/msp.h"
+#include "fc/rc_modes.h"
+#include "common/maths.h"
+
+
+
+static rcReadRawDataFnPtr rxMspOverrideRcReadRawFn;
+static rcFrameStatusFnPtr rxMspOverrideRcFrameStatusFn;
+static uint32_t rxMspOverridePassthroughChannels;
+
+
+static uint16_t rxMspOverrideReadRawRC(const rxRuntimeState_t *rxRuntimeState, uint8_t chan)
+{
+    uint16_t mspSample = rxMspReadRawRC(rxRuntimeState, chan);
+    uint16_t overrideSample = (*rxMspOverrideRcReadRawFn)(rxRuntimeState, chan);
+    bool passthrough = (1 << chan) & rxMspOverridePassthroughChannels;
+
+    if (IS_RC_MODE_ACTIVE(BOXMSPOVERRIDE) && !passthrough) {
+        return mspSample;
+    } else {
+        return overrideSample;
+    }
+}
+
+static uint8_t rxMspOverrideFrameStatus(rxRuntimeState_t *rxRuntimeState)
+{
+    uint8_t mspState = rxMspFrameStatus(rxRuntimeState);
+    uint8_t overrideState = (*rxMspOverrideRcFrameStatusFn)(rxRuntimeState);
+
+    // FIXME: combine status in a better way
```

Comment:
Should probably be returned based on `IS_RC_MODE_ACTIVE(BOXMSPOVERRIDE)`.

Reply:
This is what I thought as well, however what if for example you use `SBUS` and connection is lost. When in `MSP_OVERRIDE` mode I would think  that the drone should still go into failsafe as you no longer have any way to disable override.

Reply:
See https://github.com/betaflight/betaflight/pull/9915#issuecomment-657161101 for an idea on how to address this.